lib/propolis/src/hw/virtio/block.rs

```diff
@@ -76,6 +90,15 @@ impl PciVirtioBlock {
         }
     }
 
+    /// Return a combined queue + request id for associating an I/O
+    /// request with its subsequent completion.
+    ///
+    /// Note: this is only used for debugging purposes.
+    fn next_req_id(&self) -> u32 {
+        // We only support a single queue today
+        (0 << 16) | self.next_request_id.fetch_add(1, Ordering::Relaxed) as u32
```

Comment:
We could pull the index out of the `Chain` and use that as the ID

Reply:
i did try that and found the space of ids is much smaller that way (e.g. booting alpine would issue some ~1300 reads but reuse `idx` 0-8). Not a huge issue, but figured if a trace drops some events you might end up correlating the wrong enqueue/complete probes.

Reply:
Ah, so this is just the descriptor ID.  We could have `VirtQueue::pop_avail` (and `VqAvail::read_next_avail`) emit the index in the avail ring in addition to the descriptor ID.  Then you'd get a range as large as virtqueue, and be able to correlate that back to memory contents if you paused the VM to inspect it.

We could also leave as-written.  What do you think?

Reply:
Updated to use the avail idx